src/main/java/de/zalando/aruha/nakadi/enrichment/Enrichment.java

```diff
@@ -0,0 +1,49 @@
+package de.zalando.aruha.nakadi.enrichment;
+
+import de.zalando.aruha.nakadi.domain.EnrichmentStrategyDescriptor;
+import de.zalando.aruha.nakadi.domain.EventCategory;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.EnrichmentException;
+import de.zalando.aruha.nakadi.exceptions.InvalidEventTypeException;
+import org.json.JSONObject;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Enrichment {
+    private final EnrichmentsRegistry registry;
+
+    public Enrichment(final EnrichmentsRegistry registry) {
+        this.registry = registry;
+    }
+
+    public void validate(final EventType eventType) throws InvalidEventTypeException {
```

Comment:
Ok, I see, so it's really validated.
About the exposing externals - that's the trade-off of having enum and automatic spring validation :)